ray-operator/controllers/ray/common/pod_test.go

```diff
@@ -271,9 +308,27 @@ func TestBuildPod(t *testing.T) {
 	// testing worker pod
 	worker := cluster.Spec.WorkerGroupSpecs[0]
 	podName = cluster.Name + DashSymbol + string(rayiov1alpha1.WorkerNode) + DashSymbol + worker.GroupName + DashSymbol + utils.FormatInt32(0)
-	podTemplateSpec = DefaultWorkerPodTemplate(*cluster, worker, podName, svcName)
-	pod = BuildPod(podTemplateSpec, rayiov1alpha1.WorkerNode, worker.RayStartParams, svcName, nil)
+	podTemplateSpec = DefaultWorkerPodTemplate(*cluster, worker, podName, svcName, "6379")
+	pod = BuildPod(podTemplateSpec, rayiov1alpha1.WorkerNode, worker.RayStartParams, svcName, "6379", nil)
+
+	// Check RAY_ADDRESS env
+	foundEnv = false
+	expectedResult = "raycluster-sample-head-svc:6379"
+	for _, env := range pod.Spec.Containers[0].Env {
+		if env.Name == RAY_ADDRESS {
+			actualResult = env.Value
+			if !(actualResult == expectedResult) {
+				t.Fatalf("Expected `%v` but got `%v`", expectedResult, actualResult)
+			}
+			foundEnv = true
+			break
+		}
+	}
+	if !foundEnv {
+		t.Fatalf("Couldn't find RAY_ADDRESS env on head pod.")
```

Comment:
head pod -> worker pod?

Also, can factor this code out into a function so we can re-use it for both head and worker pods? It looks the same :)

It might also be possible to do this a bit simpler, like checking `pod.Spec.Containers[0].Env[RAY_ADDRESS]`, but not super sure about this :)

Reply:
Will fix typo.
Will factor out into function. I enjoy Vim yank and put a bit too much :)

Reply:
`pod.Spec.Containers[0].Env[RAY_ADDRESS]`
doesn't work unfortunately because `Env` is a list of Key Value pairs rather than a map